### ISSUE TITLE
[Feature Request]: Support Selective Activation Checkpoint

### ISSUE THREAD

**Issue by xingyaoww** [pytorch/torchtune#2101]
We may ends up with higher throughput if torchtune can support [selective activation checkpointing](https://pytorch.org/blog/maximizing-training/#selective-activation-checkpointing).

**Comment by felipemello1** [comment]
FYI, if you go down this rabbit role, we currently checkpoint on [every transformer block](https://github.com/pytorch/torchtune/blob/32e265d5749fd592711a03247486eafa6c898d94/recipes/full_finetune_single_device.py#L429). An interesting experiment would be to find the most expensive operations and keep their activations, while discard those that are fast to compute, like normalization.

**Comment by xingyaoww** [comment]
Hey @ebsmothers -- Thanks a lot for this info! It haven't make into the 0.4.0 release yet right? Will try to test it when i got time :)

**Comment by ebsmothers** [comment]
Oh @xingyaoww it should be in there, at least for the distributed full finetune. We actually landed the prototype feature a while ago so that should definitely be in 0.4.0. Please do let me know if you find it useful! If so happy to open an issue to enable it in some of our other recipes
